### PR TITLE
pythonPackages.executing: 0.8.2 -> 1.2.0

### PR DESCRIPTION
**Pull request by apeschar**
###### Description of changes

Adds support for Python 3.11.

Commits: https://github.com/alexmojaki/executing/compare/v0.8.2...v1.2.0

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/executing/default.nix [hunk 1]
```diff
@@ -3,18 +3,20 @@
...
 , setuptools-scm
+, pytestCheckHook
+, littleutils
 }:
 
 buildPythonPackage rec {
   pname = "executing";
-  version = "0.8.2";
+  version = "1.2.0";
   format = "setuptools";
```
Comment (tjni) [inline pkgs/development/python-modules/executing/default.nix:13]: Would you mind updating this to `format = "pyproject";`?

File: pkgs/development/python-modules/executing/default.nix [hunk 2]
```diff
@@ -25,10 +27,10 @@ buildPythonPackage rec {
     export SETUPTOOLS_SCM_PRETEND_VERSION="${version}"
   '';
 
-  # Tests appear to run fine (Ran 22 tests in 4.076s) with setuptoolsCheckPhase
-  # but crash with pytestCheckHook
   checkInputs = [
+    pytestCheckHook
     asttokens
+    littleutils
   ];
```
Comment (tjni) [inline pkgs/development/python-modules/executing/default.nix:34]: Per [this line in setup.cfg](https://github.com/alexmojaki/executing/blob/master/setup.cfg#L37), let's add:

```
] ++ lib.optionals (pythonAtLeast "3.11") [
  rich
];
```

**Review by tjni — COMMENTED** [review]
Thank you for making this change!

Can you retarget this to the staging branch, following [these instructions](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#rebasing-between-branches-ie-from-master-to-staging)?

I'm not sure if `executing` itself triggers enough rebuilds to require it, but its dependency asttokens seems to have.

This will be needed by https://github.com/NixOS/nixpkgs/issues/198533 (for general context).

**Comment by apeschar** [comment]
Not having my best day, maybe I should have read those rebase instructions first.

Sorry for inviting everyone.

**Review by tjni — APPROVED** [review]

**Comment by dotlambda** [comment]
Please squash your commits next time!!!

**Pull request merged**